Automattic/jetpack
# Pay with PayPal: Save product in Customize > Widgets

Fixes https://github.com/Automattic/wp-calypso/issues/61319

#### Changes proposed in this Pull Request:
Ensures that Simple Payment products created via the Pay with PayPal blocks inside Customize > Widgets are saved after changing the widgets, so the block is displayed in both the preview and frontend after doing any change to the block.

The only problem is that since changes need to be reflected in real time the Customizer preview, I needed to validate the attributes after every change (as opposed to the behavior in a normal post editor where attributes are only validated after deselecting the block), so users will get validation errors as soon as they start filling the block.

#### Jetpack product discussion
N/A

#### Does this pull request change what data or activity we track or use?
No

#### Testing instructions:
- Apply these changes to a Jetpack site, WP.com Simple site, or WP.com Atomic site.
- Switch to a non-FSE theme like Twenty Twenty-One.
- Go to Appearance > Customize > Widgets.
- Insert a Pay with PayPal block.
- Select a widget area.
- Make sure the block shows up in the preview after entering correct values.
- Save the changes.
- Visit the frontend.
- Make sure the block shows up in the frontend.
